src/Particle/MCCoords.cpp

```diff
@@ -15,11 +15,36 @@ namespace qmcplusplus
 {
 void MCCoords<CoordsType::POS>::resize(const std::size_t size) { positions.resize(size); }
 
+void MCCoords<CoordsType::POS>::getSubset(const std::size_t offset,
```

Comment:
I mentioned this to @ye-luo on slack a while back, but he said that effects the random numbers and would impact a few tests. So I added the getSubset stuff so we can keep the current functionality. 

I didn't actually try putting it into the loop so I'm not sure which tests (if any) would break.

Reply:
changed to return by value

Reply:
I restored the out argument. Since the out argument passed in is reused. This way avoids frequent allocate and deallocate memory.